### ISSUE TITLE
Cannot build for MacOS if I also include shared_preferences plugin

### ISSUE THREAD

**Issue by kaciula** [yeniel/native_shared_preferences#18]
When building for MacOS I get `error: ambiguous use of 'register(with:)'` and this is because the GeneratedPluginRegistrant.swift file contains 2 lines to register a plugin with the same name: SharedPreferencesPlugin

The name of this plugin should be different than the one from shared_preferences so that they can be both included when building for Mac.

![Screenshot 2024-09-05 at 11 27 55](https://github.com/user-attachments/assets/ff42538f-22b9-45f6-995c-666aec83a368)

**Comment by JetA2** [comment]
I have the same issue. I suspect it's because `native_shared_preferences` depends on `shared_preferences_macos` while the `shared_preferences` package depends on the new unified `shared_preferences_foundation` which is for both MacOS and iOS. The generated plugin registrant then tries to register both.

**Comment by kaciula** [comment]
@JetA2 Have you figured out a workaround to be able to build for MacOS using both plugins?

**Comment by JetA2** [comment]
> @JetA2 Have you figured out a workaround to be able to build for MacOS using both plugins?

Unfortunately not.

**Comment by Aehawk** [comment]
@kaciula @JetA2 Have you found any solution to this problem?

**Comment by kaciula** [comment]
@Aehawk No. I didn't need this plugin anymore. But what you can do is use the latest versions of shared_preferences plugin that allows you to get all native preferences without any prefix.

**Comment by Aehawk** [comment]
@kaciula Oh thank you. Actually after I researched a bit I found a workaround with platform channels. I'm invoking the key from the flutter side and fetching UserDefault data from appdelegate.swift then returning value of key to the flutter. I cannot use latest shared_preferences package due to my project's flutter version is a bit old.
